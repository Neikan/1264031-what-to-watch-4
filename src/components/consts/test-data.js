export const Promo = {
  TITLE: `The Grand Budapest Hotel poster`,
  GENRE: `Drama`,
  DATE: 2014
};

export const MOVIES = [
  {
    id: `mock-1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`
  }, {
    id: `mock-2`,
    title: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody`
  }, {
    id: `mock-3`,
    title: `Macbeth`,
    poster: `macbeth`
  }
];

