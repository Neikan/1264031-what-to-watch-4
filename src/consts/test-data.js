const MOVIE_POSTER = `the-grand-budapest-hotel-poster`;

export const LogoPosition = {
  HEADER: `header`,
  FOOTER: `footer`
};


export const MOVIES = [
  {
    id: `mock-1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: MOVIE_POSTER,
    cover: `fantastic-beasts-the-crimes-of-grindelwald`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/f/fd/%22Night_After_Night%22_Trailer_1932.webm/%22Night_After_Night%22_Trailer_1932.webm.480p.vp9.webm`,
    rating: {
      score: 4.6,
      level: `Normal`,
      countVotes: 240,
    },
    director: `Крис МакКей`,
    actors: `Крис О’Доннелл, Джек Николсон`,
    duration: 120,
    year: 2015,
    genre: `Documentary`,
    description: `Восемь лет назад Бэтмен растворился в ночи, превратившись из героя в беглеца.`,
    reviews: []
  }, {
    id: `mock-2`,
    title: `Bohemian Rhapsody`,
    poster: MOVIE_POSTER,
    cover: `bohemian-rhapsody`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/c/cd/12_Angry_Men_%281957%29_-_Trailer.webm/12_Angry_Men_%281957%29_-_Trailer.webm.480p.vp9.webm`,
    rating: {
      score: 9.1,
      level: `Very good`,
      countVotes: 10,
    },
    director: `Крис МакКей`,
    actors: `Арнольд Шварценеггер, Кристиан Бэйл`,
    duration: 320,
    year: 1999,
    genre: `Crime`,
    description: `Бэтмен поднимает ставки в войне с криминалом. С помощью лейтенанта Джима Гордона и прокурора Харви Дента он намерен очистить улицы от преступности, отравляющей город. Сотрудничество оказывается эффективным, но скоро они обнаружат себя посреди хаоса, развязанного восходящим криминальным гением, известным испуганным горожанам под именем Джокер.`,
    reviews: [{
      id: `reviews-9`,
      text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
      author: `Lickona Matthew`,
      date: `December 2, 2016`,
      rating: 1.2
    }]
  }, {
    id: `mock-3`,
    title: `Macbeth`,
    poster: MOVIE_POSTER,
    cover: `macbeth`,
    preview: `https://upload.wikimedia.org/wikipedia/commons/transcoded/c/ca/2001_A_Space_Odyssey_%281968%29_-_Trailer.webm/2001_A_Space_Odyssey_%281968%29_-_Trailer.webm.360p.vp9.webm`,
    rating: {
      score: 6.0,
      level: `Good`,
      countVotes: 1240,
    },
    director: `Джоэл Шумахер`,
    actors: ` Ким Бейсингер, Том Харди, Кэнта Миякэ, Марион Котийяр`,
    duration: 20,
    year: 2020,
    genre: `Horror`,
    description: `Приняв на себя вину за смерть прокурора Харви Дента, он пожертвовал всем. Вместе с комиссаром Гордоном они решили, что так будет лучше для всех.`,
    reviews: [{
      id: `reviews-1`,
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      author: `Kate Muir`,
      date: `December 24, 2016`,
      rating: 8.9
    }, {
      id: `reviews-2`,
      text: `Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`,
      author: `Bill Goodykoontz`,
      date: `November 18, 2015`,
      rating: 5.0
    }, {
      id: `reviews-3`,
      text: `I didn't find it amusing, and while I can appreciate the creativity, it's an hour and 40 minutes I wish I could take back.`,
      author: `Amanda Greever`,
      date: `November 18, 2015`,
      rating: 3.3
    }, {
      id: `reviews-4`,
      text: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
      author: `Matthew Lickona`,
      date: `December 20, 2016`,
      rating: 7.2
    }, {
      id: `reviews-5`,
      text: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
      author: `Paula Fleri-Soler`,
      date: `December 20, 2016`,
      rating: 10
    }]
  }
];


export const GENRES = [
  {
    id: `g1`,
    title: `All genres`,
    isActive: true
  }, {
    id: `g2`,
    title: `Comedies`,
    isActive: false
  }, {
    id: `g3`,
    title: `Crime`,
    isActive: false
  }
];


export const MarkupElement = {
  PROMO_BTN_PLAY: `btn--play`,
  PROMO_BTN_ADD_TO_LIST: `btn--list`,
  MOVIE_CARD: `small-movie-card`,
  MOVIE_CARD_TITLE: `small-movie-card__title`,
  MOVIE_CARD_COVER: `small-movie-card__image`,
  GENRE_LINK: `catalog__genres-link`,
  MOVIE_TAB: `movie-nav__item`,
  BTN_SHOW_MORE: `catalog__button`
};
