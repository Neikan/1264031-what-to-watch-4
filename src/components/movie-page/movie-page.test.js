// Импорт библиотек
import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";

// Импорт компонентов
import MoviePage from "./movie-page.jsx";

// Импорт типов, констант, утилит
import {MOVIES, MovieTabList, AuthStatus} from "../../consts/test-data.js";
import NameSpace from "../../store/name-space.js";


const mockStore = configureStore([]);


describe(`Test MoviePage component`, () => {
  const store = mockStore({
    [NameSpace.DATUM_USER]: {
      authStatus: AuthStatus.NO_AUTH,
      user: {
        id: ``,
        email: ``,
        name: ``,
        avatarUrl: ``
      },
    }
  });

  test(`MoviePage component is created and rendered correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Provider store={store}>
            <MoviePage
              authStatus={AuthStatus.NO_AUTH}
              movie={MOVIES[0]}
              movies={MOVIES}
              countShowedMovies={4}
              onMovieSelect={() => {}}
              onMovieChangePlaying={() => {}}
              selectedTab={MovieTabList.OVERVIEW}
              onTabSelect={() => {}}
            />
          </Provider>
        </BrowserRouter>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
