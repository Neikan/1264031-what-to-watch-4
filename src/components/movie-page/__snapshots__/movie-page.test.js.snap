// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MoviePage component MoviePage component is created and rendered correctly 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  >
    <div
      className="movie-card__hero"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Fantastic Beasts: The Crimes of Grindelwald"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom3.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header movie-card__head"
      >
        <div
          className="logo"
        />
        <div
          className="user-block"
        />
      </header>
      <div
        className="movie-card__wrap"
      >
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            >
              Documentary
            </span>
            <span
              className="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            className="movie-card__buttons"
          >
            <button
              className="btn btn--list movie-card__button"
              onClick={[Function]}
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="movie-card__wrap movie-card__translate-top"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="Fantastic Beasts: The Crimes of Grindelwald"
            height="327"
            src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/moonrise-kingdom1.jpg"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <nav
            className="movie-nav movie-card__nav"
          >
            <ul
              className="movie-nav__list"
            >
              <li
                className="movie-nav__item movie-nav__item--active"
                onClick={[Function]}
              />
              <li
                className="movie-nav__item"
                onClick={[Function]}
              />
              <li
                className="movie-nav__item visually-hidden"
                onClick={[Function]}
              />
            </ul>
          </nav>
          <div
            className="movie-card__text movie-card__row"
          >
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Director
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Крис МакКей
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Starring
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Крис О’Доннелл, Джек Николсон
                </span>
              </p>
            </div>
            <div
              className="movie-card__text-col"
            >
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Run Time
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  120
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Genre
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  Documentary
                </span>
              </p>
              <p
                className="movie-card__details-item"
              >
                <strong
                  className="movie-card__details-name"
                >
                  Released
                </strong>
                <span
                  className="movie-card__details-value"
                >
                  2015
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <div
    className="page-content"
  >
    <section
      className="catalog catalog--like-this"
    >
      <h2
        className="catalog__title"
      >
        More like this
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/gangs_of_new_york2.jpg"
              type="video/webm"
              width="280"
            >
              Sorry, your browser doesn\`t support embedded videos
            </video>
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          />
        </article>
        <article
          className="small-movie-card catalog__movies-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <div
            className="small-movie-card__image"
          >
            <video
              height="175"
              poster="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/pulp-fiction2.jpg"
              type="video/webm"
              width="280"
            >
              Sorry, your browser doesn\`t support embedded videos
            </video>
          </div>
          <h3
            className="small-movie-card__title"
            onClick={[Function]}
          />
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      />
      <div
        className="copyright"
      >
        <p>
          © 2020 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
